apps/wakunode2/wakunode2_validator_signed.nim

```diff
@@ -49,23 +51,53 @@ proc withinTimeWindow*(msg: WakuMessage): bool =
     return true
   return false
 
-proc addSignedTopicValidator*(w: WakuRelay, topic: PubsubTopic, publicTopicKey: SkPublicKey) =
-  debug "adding validator to signed topic", topic=topic, publicTopicKey=publicTopicKey
+proc getKeyFromTopic*(protectedTopic: PubsubTopic): Result[string, string] =
+  # example: /waku/2/signed:0x2ea1f2ec2da14e0e3118e6c5bdfa0631785c76cd/proto
+  # see https://rfc.vac.dev/spec/57/#dos-protection
+  let topicTag = "signed"
+  let parts = protectedTopic.split("/")
+  if parts.len != 5:
+    return err("invalid topic format for a signed protected topic, missing fields")
 
-  proc validator(topic: string, message: messages.Message): Future[errors.ValidationResult] {.async.} =
+  let topicName = parts[3]
+  let topicParts = topicName.split(":")
+  if topicParts.len != 2:
+    return err("invalid topic format for a signed protected topic, : missing")
+
+  if topicTag notin topicName:
+    return err("invalid topic format expected signed: tag")
+
+  let address = topicParts[1]
+  if not address.startsWith("0x") or address.len != 42:
+    return err("invalid address, expected 0x prefix and size of 42 chars")
+
+  return ok(address)
```

Comment:
```suggestion
  return ok(address.toLowerAscii)
```

Reply:
Maybe it makes sense to return an error if the address is not in lowercase at all?